src/cmd-build

```diff
@@ -204,26 +198,29 @@ cat > tmp/meta.json <<EOF
 EOF
 # Merge all the JSON; note that we want ${composejson} first
 # since we may be overriding data from a previous build.
-cat ${composejson} tmp/meta.json ${commitmeta_input_json} | jq -s add > meta.json
+cat "${composejson}" tmp/meta.json "${commitmeta_input_json}" | jq -s add > meta.json
 
 # Clean up our temporary data
 rm tmp -rf
 # Back to the toplevel build directory, so we can rename this one
-cd ${workdir}/builds
+cd "${workdir}"/builds
 # We create a .build-commit file to note that we're in the
 # middle of a "commit".  This may be useful in the future
 # for having things be transactional.  If for example we
 # were interrupted between the rename() and linkat() below,
 # things would be inconsistent and future builds would fail
 # on the `mv`.
 touch .build-commit
-mv -T ${tmp_builddir} ${buildid}
+mv -T "${tmp_builddir:?}" "${buildid}"
```

Comment:
Hmm, why the explicit `:?` here?  We're using `set -u` so that behavior applies everywhere right?

Reply:
(It doesn't hurt though, I'm fine leaving this as is)

Reply:
https://github.com/koalaman/shellcheck/wiki/SC2154

I used the guidance at the bottom:

`f you know for a fact that the variable is set, you can use ${var:?} to fail if the variable is unset (or empty), or explicitly initialize/declare it with var="" or declare var.`

Reply:
Ah, so if I have the `shellcheck` directive at the top of the file `# shellcheck source=src/cmdlib.sh`, this warning doesn't get thrown any more.  Leaving the `:?` doesn't seem to hurt it either.

Reply:
Ah ok!